src/cpp/src/lm_encoding.cpp

```diff
@@ -216,13 +223,7 @@ std::pair<EncodedResults, int32_t> get_lm_encoded_results(
                                                     active_sequence_groups.end(),
                                                     get_active_sequence_groups),
                                     active_sequence_groups.end());
-    }
-
-    if (streamer_ptr) {
-        int64_t out_token = sequence_groups.at(0).get()->operator[](0)->get_generated_ids().back();
```

Comment:
aren't we missing the last generated token by deleting this ?

Reply:
No, because the cycle was changed: `while{} -> do while{}`